backend/matching-service/src/services/rabbitmq.service.ts

```diff
@@ -0,0 +1,82 @@
+import client, { Connection, Channel } from 'amqplib'
+
+import config from '../common/config.util'
+import { IUserQueueMessage } from '../types/IUserQueueMessage'
+import logger from '../common/logger.util'
+
+class RabbitMQConnection {
+    connection!: Connection
+    channel!: Channel
+    private connected!: boolean
+
+    async connect() {
+        if (this.connected && this.channel) return
+        else this.connected = true
+
+        try {
+            logger.info(`[Init] Connecting to Rabbit-MQ Server`)
+            this.connection = await client.connect({
+                protocol: 'amqp',
+                hostname: config.RMQ_HOST,
+                port: 5672,
+                username: config.RMQ_USER,
+                password: config.RMQ_PASSWORD,
+            })
+
+            logger.info(`[Init] Rabbit MQ Connection is ready`)
+
+            this.channel = await this.connection.createChannel()
+
+            logger.info(`[Init] Created RabbitMQ Channel successfully`)
+        } catch (error) {
+            logger.error(error)
+            logger.error(`[Error] Not connected to MQ Server`)
```

Comment:
you set connected to true in line 14 but then if the connection fails you dont set it back to false, is this expected behaviour? I would assume that you would set it back to false so that you could retry

Reply:
ok will change this